airbyte-cdk/python/airbyte_cdk/sources/streams/concurrent/partition_enqueuer.py

```diff
@@ -34,7 +39,9 @@ def generate_partitions(self, stream: AbstractStream) -> None:
         """
         try:
             for partition in stream.generate_partitions():
-                self._queue.put(partition)
```

Comment:
It feels like there is some logic associated with how we access the queue. I wonder if it would make sense to regroup this in a class. We have similar logic in PartitionReader for example

Reply:
for sure! I actually ran into another bottleneck preventing the sync from succeeding in a reasonable amount of time. I'll clean this up before asking for a formal review